onedal/spmd/linear_model/incremental_linear_model.py

```diff
@@ -0,0 +1,97 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import numpy as np
+
+from daal4py.sklearn._utils import get_dtype
+
+from ..._device_offload import support_usm_ndarray
+from ...common.hyperparameters import get_hyperparameters
+from ...datatypes import _convert_to_supported, to_table
+from ...linear_model import (
+    IncrementalLinearRegression as IncrementalLinearRegression_nonSPMD,
+)
+from ...utils import _check_X_y, _num_features
+from .._base import BaseEstimatorSPMD
+
+
+class IncrementalLinearRegression(BaseEstimatorSPMD, IncrementalLinearRegression_nonSPMD):
+    def _reset(self):
+        self._partial_result = super(
+            IncrementalLinearRegression_nonSPMD, self
+        )._get_backend("linear_model", "regression", "partial_train_result")
+
+    @support_usm_ndarray()
+    def partial_fit(self, X, y, queue=None):
+        """
+        Computes partial data for linear regression
+        from data batch X and saves it to `_partial_result`.
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training data batch, where `n_samples` is the number of samples
+            in the batch, and `n_features` is the number of features.
+
+        y: array-like of shape (n_samples,) or (n_samples, n_targets) in
+            case of multiple targets
+            Responses for training data.
+
+        queue : dpctl.SyclQueue
+            If not None, use this queue for computations.
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        module = super(IncrementalLinearRegression_nonSPMD, self)._get_backend(
+            "linear_model", "regression"
+        )
+
+        if not hasattr(self, "_queue"):
+            self._queue = queue
+        policy = super(IncrementalLinearRegression_nonSPMD, self)._get_policy(queue, X)
+
+        X, y = _convert_to_supported(policy, X, y)
+
+        if not hasattr(self, "_dtype"):
+            self._dtype = get_dtype(X)
+            self._params = self._get_onedal_params(self._dtype)
+
+        y = np.asarray(y).astype(dtype=self._dtype)
+        self._y_ndim_1 = y.ndim == 1
+
+        X, y = _check_X_y(X, y, dtype=[np.float64, np.float32], accept_2d_y=True)
+
+        self.n_features_in_ = _num_features(X, fallback_1d=True)
+        X_table, y_table = to_table(X, y)
+        hparams = get_hyperparameters("linear_regression", "train")
```

Comment:
Why is any of this needed? Shouldn't this be covered by batch estimator call? There are no data preprocessing function calls in other spmd functionality.

Reply:
this method overrides the regular one, that's why data preprocessing is necessary here as well